### PR TITLE
Diverse log improvements, comment additions and small refactors.

### PR DESCRIPTION
**Pull request by nalepae**
**What type of PR is this?**
Other

**What does this PR do? Why is it needed?**
This PR pulls into develop a lot of non functional changes, unrelated to peerDAS, but currently in the `peerDAS` branch.

**Other notes for review**
Please read commit by commit.

**Acknowledgements**
- [x] I have read [CONTRIBUTING.md](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md).
- [x] I have made an appropriate entry to [CHANGELOG.md](https://github.com/prysmaticlabs/prysm/blob/develop/CHANGELOG.md).
- [x] I have added a description to this PR with sufficient context for reviewers to understand this PR.

### CODE REVIEW

File: beacon-chain/p2p/utils.go [hunk 1]
```diff
@@ -173,3 +179,26 @@ func verifyConnectivity(addr string, port uint, protocol string) {
 		}
 	}
 }
+
+func ConvertPeerIDToNodeID(pid peer.ID) (enode.ID, error) {
```
Comment (terencechain) [inline beacon-chain/p2p/utils.go:183]: Should we add a godoc comment?

Comment (nalepae) [reply beacon-chain/p2p/utils.go:184]: Sure, done in 85264bb21141ab238497fb23e985631d0f602799.

File: CHANGELOG.md [hunk 2]
```diff
@@ -90,6 +90,7 @@ The format is based on Keep a Changelog, and this project adheres to Semantic Ve
 - Fix panic on attestation interface since we call data before validation
 - corrects nil check on some interface attestation types
 - temporary solution to handling electra attesation and attester_slashing events. [pr](14655)
+- Diverse log improvements, comment additions and logs improment.
```
Comment (saolyn) [inline CHANGELOG.md:93]:
```suggestion
- Diverse log improvements and comment additions.
```

File: beacon-chain/sync/rpc_beacon_blocks_by_range.go [hunk 3]
```diff
@@ -82,12 +91,19 @@ func (s *Service) beaconBlocksByRangeRPCHandler(ctx context.Context, msg interfa
 		}
 		rpcBlocksByRangeResponseLatency.Observe(float64(time.Since(batchStart).Milliseconds()))
 	}
+
 	if err := batch.error(); err != nil {
-		log.WithError(err).Debug("error in BlocksByRange batch")
-		s.writeErrorResponseToStream(responseCodeServerError, p2ptypes.ErrGeneric.Error(), stream)
+		log.WithError(err).Debug("Serving block by range request - BlocksByRange batch")
+
+		// If we hit a rate limit, the error response has already been written, and the stream is already closed.
```
Comment (saolyn) [inline beacon-chain/sync/rpc_beacon_blocks_by_range.go:98]:
```suggestion
		// If a rate limit is hit, it means an error response has already been sent and the stream has been closed.
```

File: beacon-chain/sync/rpc_blob_sidecars_by_range.go [hunk 4]
```diff
@@ -114,7 +115,12 @@ func (s *Service) blobSidecarsByRangeRPCHandler(ctx context.Context, msg interfa
 	}
 	if err := batch.error(); err != nil {
 		log.WithError(err).Debug("error in BlobSidecarsByRange batch")
-		s.writeErrorResponseToStream(responseCodeServerError, p2ptypes.ErrGeneric.Error(), stream)
+
+		// If we hit a rate limit, the error response has already been written, and the stream is already closed.
```
Comment (saolyn) [inline beacon-chain/sync/rpc_blob_sidecars_by_range.go:119]:
```suggestion
		// If a rate limit is hit, it means an error response has already been sent and the stream has been closed.
```

File: beacon-chain/sync/rpc_ping.go [hunk 5]
```diff
@@ -16,127 +16,191 @@ import (
 )
 
 // pingHandler reads the incoming ping rpc message from the peer.
+// If the peer's sequence number is higher than the one we have in our store for it,
+// we send a METADATA request to the peer to get its latest metadata, and store it.
+// This function is actually poorly named, since it does more than just reading a ping message.
```
Comment (saolyn) [inline beacon-chain/sync/rpc_ping.go:21]:
```suggestion
// If the peer's sequence number is higher than the one stored locally, 
// a METADATA request is sent to the peer to retrieve and update the latest metadata.
// Note: This function is misnamed, as it performs more than just reading a ping message.
```

**Review by terencechain — APPROVED** [review]

**Review by nisdas — APPROVED** [review]

**Pull request merged**